### PR TITLE
MolGANEncoderLayer Porting

### PR DESCRIPTION
**Pull request by shreyasvinaya**
## Description
Torch implementation of MolGANEncoderLayer
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/tests/test_molgan_layers.py [hunk 1]
```diff
@@ -235,3 +235,118 @@ def test_multigraph_convolution_layer_values():
...
+    torch.manual_seed(21)
+    tf_weights = np.load(
+        'deepchem/models/tests/assets/molgan_encoder_layer_weights.npy',
+        allow_pickle=True).item()
+    torch_model_encoder = MolGANEncoderLayer(units=units,
+                                             nodes=nodes,
+                                             edges=edges,
+                                             name='layer1')
+
+    x = 12
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_molgan_layers.py:286]: Why is x set to 12 here? In general we should avoid magic numbers without explanations in tests

File: deepchem/models/torch_models/tests/test_molgan_layers.py [hunk 2]
```diff
@@ -235,3 +235,118 @@ def test_multigraph_convolution_layer_values():
...
+    assert layer.graph_convolution_units == (first_convolution_unit,
+                                             second_convolution_unit)
+    assert layer.auxiliary_units == aggregation_unit
+    assert layer.activation == torch.tanh
+    assert layer.edges == 5
+    assert layer.dropout_rate == 0.0
+
+
+@pytest.mark.torch
+def test_graph_encoder_layer_values():
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_molgan_layers.py:267]: This test is getting very complicated. Can you add comments explaning the logic of what the full test is doing?

**Review by rbharath — COMMENTED** [review]
The PR is mostly good but I have a few requests for additional comments below

File: deepchem/models/torch_models/tests/test_molgan_layers.py (hunk 1 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/tests/test_molgan_layers.py:286]: it was the starting number of the dense layers in the weights, have added a comment explaining the same

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**